### PR TITLE
Remove forEach, ElementVisitor and ElementReader.

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: tape/src/main/java/com/squareup/tape/FileObjectQueue.java [hunk 1]
```diff
@@ -138,15 +135,9 @@ public final void close() {
   @Override public void setListener(final Listener<T> listener) {
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/FileObjectQueue.java]: Could this go out in a separate PR too?

Comment (JakeWharton) [inline tape/src/main/java/com/squareup/tape/FileObjectQueue.java]: Maybe. I'll have to look at internal use.

Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/FileObjectQueue.java]: Looks very sparsely used on Github https://github.com/search?utf8=%E2%9C%93&q=com.squareup.tape.FileObjectQueue+Listener&type=Code&ref=searchresults, mostly just in forks of Tape.

**Comment by JakeWharton** [comment]
Nuke `FileException` next?

File: tape/src/main/java/com/squareup/tape/FileObjectQueue.java [hunk 2]
```diff
@@ -138,15 +135,9 @@ public final void close() {
...
-          @Override public boolean read(InputStream in, int length) throws IOException {
-            byte[] data = new byte[length];
-            in.read(data, 0, length);
-
-            listener.onAdd(FileObjectQueue.this, converter.from(data));
-            return true;
-          }
-        });
+        for (byte[] data : queueFile) {
+          listener.onAdd(FileObjectQueue.this, converter.from(data));
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/FileObjectQueue.java]: Ah this snuck through :(
